contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+        if (tokenId != 0 && lpTokenDepositedToGauge) {
+            clGauge.withdraw(tokenId);
+            lpTokenDepositedToGauge = false;
+        }
+        _;
+        if (tokenId != 0 && !lpTokenDepositedToGauge) {
+            (
+                uint256 _wethPositionBalance,
+                uint256 _oethbPositionBalance
+            ) = getPositionPrincipal();
+
+            /**
+             * It can happen that a withdrawal (or a full withdrawal) transactions would
+             * remove all of the liquidity from the token with a NFT token still existing.
+             * In that case the token can not be staked into the gauge, as some liquidity
+             * needs to be added to it first.
+             */
+            if (_wethPositionBalance + _oethbPositionBalance > 0) {
+                positionManager.approve(address(clGauge), tokenId);
+                clGauge.deposit(tokenId);
+                lpTokenDepositedToGauge = true;
+            }
+        }
+    }
+
+    /// @notice the constructor
+    /// @param _stratConfig the basic strategy configuration
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _oethbAddress Address of the Erc20 OETHb Token contract
+    /// @param _swapRouter Address of the Aerodrome Universal Swap Router
+    /// @param _nonfungiblePositionManager Address of position manager to add/remove
+    ///         the liquidity
+    /// @param _clPool Address of the Aerodrome concentrated liquidity pool
+    /// @param _clGauge Address of the Aerodrome slipstream pool gauge
+    /// @param _sugarHelper Address of the Aerodrome Sugar helper contract
+    /// @param _lowerBoundingTick Smaller bounding tick of our liquidity position
+    /// @param _upperBoundingTick Larger bounding tick of our liquidity position
+    /// @param _tickClosestToParity Tick that is closer to 1:1 price parity
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _wethAddress,
+        address _oethbAddress,
+        address _swapRouter,
+        address _nonfungiblePositionManager,
+        address _clPool,
+        address _clGauge,
+        address _sugarHelper,
+        int24 _lowerBoundingTick,
+        int24 _upperBoundingTick,
+        int24 _tickClosestToParity
+    ) InitializableAbstractStrategy(_stratConfig) {
+        WETH = _wethAddress;
+        OETHb = _oethbAddress;
+        swapRouter = ISwapRouter(_swapRouter);
+        positionManager = INonfungiblePositionManager(
+            _nonfungiblePositionManager
+        );
+        clPool = ICLPool(_clPool);
+        clGauge = ICLGauge(_clGauge);
+        helper = ISugarHelper(_sugarHelper);
+        sqrtRatioX96TickLower = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(
+            _lowerBoundingTick
+        );
+        sqrtRatioX96TickHigher = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(
+            _upperBoundingTick
+        );
+        sqrtRatioX96TickClosestToParity = ISugarHelper(_sugarHelper)
```

Comment:
Should there be any checks to make sure that `_tickClosestToParity` is same as either `_lowerBoundingTick` or `_upperBoundingTick`

Reply:
Great point thanks. Check added here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/661764993900cb89e47277b18f2c767aeb98f54f